src/ucs/vfs/base/vfs_obj.h

```diff
@@ -9,11 +9,40 @@
 
 #include <ucs/datastruct/string_buffer.h>
 #include <ucs/sys/compiler_def.h>
+#include <stdint.h>
 
 BEGIN_C_DECLS
 
 /* This header file defines API for manipulating VFS object tree structure */
 
+
+/* Defines type of primitive variables */
+typedef enum {
+    /* Basic type definitions */
+    UCS_VFS_TYPE_POINTER,
+    UCS_VFS_TYPE_STRING,
+    UCS_VFS_TYPE_CHAR,
+    UCS_VFS_TYPE_SHORT,
+    UCS_VFS_TYPE_INT,
+    UCS_VFS_TYPE_LONG,
+
+    /* Type modifiers */
+    UCS_VFS_TYPE_UNSIGNED = UCS_BIT(14),
+    UCS_VFS_TYPE_HEX      = UCS_BIT(15),
```

Comment:
minor: I'd start these flags from `UCS_VFS_TYPE_LAST` to avoid potential intersection of types and modifiers in the future

Reply:
added static assert since could not find method to round_pow2 in "compile time"